internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -635,27 +660,34 @@ func NewOpenShiftAuthProxyContainer(cr *model.CryostatInstance, specs *ServiceSp
 		fmt.Sprintf("--upstream=http://localhost:%d/", constants.CryostatHTTPContainerPort),
 		fmt.Sprintf("--upstream=http://localhost:%d/grafana/", constants.GrafanaContainerPort),
 		fmt.Sprintf("--upstream=http://localhost:%d/storage/", constants.StoragePort),
-		"--cookie-secret=REPLACEME",
 		fmt.Sprintf("--openshift-service-account=%s", cr.Name),
 		"--proxy-websockets=true",
 		fmt.Sprintf("--http-address=0.0.0.0:%d", constants.AuthProxyHttpContainerPort),
-		fmt.Sprintf(
-			`--openshift-sar=[{"group":"","name":"","namespace":"%s","resource":"pods","subresource":"exec","verb":"create","version":""}]`,
-			cr.InstallNamespace,
-		),
-		fmt.Sprintf(
-			`--openshift-delegate-urls={"/":{"group":"","name":"","namespace":"%s","resource":"pods","subresource":"exec","verb":"create","version":""}}`,
-			cr.InstallNamespace,
-		),
-		"--bypass-auth-for=^/health",
 		"--proxy-prefix=/oauth2",
 	}
+	if isOpenShiftAuthProxyDisabled(cr) {
+		args = append(args, "--bypass-auth-for=.*")
+	} else {
+		args = append(args, "--bypass-auth-for=^/health")
+	}
+
+	// FIXME don't ignore this error! We're far removed from the reconciler loop - we should return this
```

Comment:
Any advice for what to do here? I didn't want to add an `error` return to this function and its caller etc. all the way back up to the reconcile loop function. Maybe it's safe to ignore because at this point the object is either a default structure we've created or else it's a reference to a CR field, which means it has already been validated and deserialized by Kubernetes to begin with - how can we have an error in serializing it?

Reply:
I would just propagate the error to the reconcile loop. A more user-friendly way could be to add the validation into our validating webhook, but that may be more RFE territory.